### DISCUSSION TITLE
study object from db to in-memory?

### DISCUSSION THREAD

**Discussion by ironv** [optuna/optuna#4112] [Q&A]
I recently switched from using in-memory study to sqlite storage for the study, where the database is stored in `/dev/shm`.  **Works really well**!  One of the things I could do previously was save the study object as a pickle and create the Optuna visualizations later in a notebook.  My VMs auto shutdown after the run and I loose the `/dev/shm` contents.  Is there a way to pull the study back into memory (from the db) after the run is done so that it can be pickled like before?  Would setting `to_storage=None` in `copy_storage` move it to in-memory?

**Comment by nzw0301** [comment]
To pickle a study on DB, loading `study and saving it as a pickle would be enough.

```python

import joblib

study = optuna.create_study(
    study_name="example-study",
    storage="sqlite:///example.db",
    load_if_exists=True
)

joblib.dump(study, "study.pkl")
```

**Reply by ironv** [reply]
I'd like to be able delete the database object which is on `/dev/shm`.  The method you recommend still seems to need that file.  Finally ended up with the following:
```
os.system(f"mv /dev/shm/{dbname} models/{dbname}")
study = optuna.load_study(study_name=clargs["TAG"], storage=f"sqlite:///models/{dbname}")
```
works but I wish there was a way to pull the completed database file into memory as though it was created with the in-memory option in the first place.
